docs/go-selectors.md

```diff
@@ -0,0 +1,120 @@
+---
+id: go-selectors
+title: Go Selectors
```

Comment:
possible alternative: "Go SDK Selectors"? to make it clear its our thing, not a Go thing